distribution/errors.go

```diff
@@ -84,7 +84,9 @@ func continueOnError(err error) bool {
 func retryOnError(err error) error {
 	switch v := err.(type) {
 	case errcode.Errors:
-		return retryOnError(v[0])
+		if len(v) != 0 {
+			return retryOnError(v[0])
+		}
```

Comment:
this check won't be necessary after https://github.com/docker/distribution/pull/1531 and distribution vendor is updated /cc @aaronlehmann